src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -124,14 +147,36 @@ const InfiniteScroll = ({
         Math.max(
           (!replace && endPage) || 0,
           multiColumn
-            ? Math.ceil(((top + height + offset) * width) / pageArea)
-            : Math.floor((top + height + offset) / pageHeight),
+            ? Math.ceil((scrollBufferBottom * width) / avgPageArea)
+            : Math.floor(scrollBufferBottom / avgPageHeight),
           show ? Math.floor(show / step) : 0,
         ),
       );
-
       if (nextBeginPage !== beginPage) setBeginPage(nextBeginPage);
       if (nextEndPage !== endPage) setEndPage(nextEndPage);
+
+      // Keep track of page heights as begin/end pages increment/decrement
```

Comment:
So the one edge case that still isn't behaving ideally is where we have the following scenario:
- `replace` && `onMore` && large variation pageHeight (pageHeights differing >20% +/-) --> specifically when scrolling past a large page of results, then "back up".  This is illustrated in the "Variable Item Height with Replace" story.

All other behaviors are solid.